packages/core/src/components/search/Filter/FilterDesktop.tsx

```diff
@@ -34,9 +40,13 @@ function FilterDesktop({
 }: FilterDesktopProps & ReturnType<typeof useFilter>) {
   const { resetInfiniteScroll, state, setState } = useSearch()
 
-  const deliveryLabel = deliverySettings?.title ?? 'Delivery'
-  const { postalCode } = sessionStore.read()
+  const context = usePage<SearchPageContext | PLPContext>()
+  const globalDeliverySettingsData =
+    context?.globalSectionsSettings?.regionalization?.deliverySettings ?? {}
+  const cmsData = deepmerge(globalDeliverySettingsData, deliverySettings ?? {})
```

Comment:
Will we need this deep merge in every component that will use this global settings? If so, maybe move this logic to a separated function?

Reply:
Great suggestion! Since it'll be used exactly like that in FilterSlider, I've extracted it to a utils file